docs/api.md

```diff
@@ -6,6 +6,171 @@ challenges presented by JavaScript and web browsers.
+
+
+#### `iodide.file.load` patterns
+
+```javascript
+
+// load a csv
+iodide.file.load('cached-data.csv', 'text', 'cachedData').then(() => {
+  cachedData = d3.csvParsed(cachedData)
+})
+
+// load a video
+iodide.file.load('gritty.mp4', 'blob', 'gritty').then(() => {
+  // load this gritty video into a <video /> tag in the report.
+  var urlCreator = window.URL || window.webkitURL;
+  var imageUrl = urlCreator.createObjectURL(gritty);
+  document.querySelector("#gritty").src = imageUrl;
+})
+
+// load a style sheet and apply the style
+iodide.file.load('corporate-report-style.css', 'css')
+
+// load some javascript helpers
+iodide.file.load('analysis-helpers.js', 'js')
+
+```
+
+
+
+### `iodide.file.delete(fileName[, deleteOptions])`
+
+Deletes the file specified by `fileName`. Returns a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) that when resolved, denotes that the file was deleted on the server.
+
+`deleteOptions` is an object with the following keys:
+
+- `throwIfDoesNotExist` (optional, default `true`): will throw an error and halt the evaluation queue if `true`.
+
+
+
+### `iodide.file.list()`
+
+Returns an Array of file names available to the current notebook.
+
+
+#### `iodide.file.list` patterns
+
+
+```javascript
+// delete the various pngs originally saved to this notebook
+iodide.file.list()
+	.filter(filename => !filename.includes('.png'))
+	.forEach(iodide.file.delete)
+```
+
+
+### `iodide.file.exists(fileName)`
+
+Returns `true` if the file `fileName` is available to the notebook, and `false` otherwise.
+
+The below code chunk, for instance, checks to see if a csv file already exists on the server. If it does, then Iodide will load that. Otherwise,
+the notebook will fetch the data, process it using a numb er of steps, and save the output to the server.
+
+
+#### `iodide.file.exists` patterns
+
+The example below loads cached data if it exists on the server, and otherwise downloads the larger dataset remotely, processes it, then caches it.
+
+```javascript
+
+const FILENAME = 'dataset.csv'
+if (iodide.file.exists(FILENAME)) {
+  // since the cached file already exists, let's load it.
+  iodide.file.load(FILENAME, 'text', 'evictionsData')
+    .then(() => {
+      evictionsData = d3.parseCsv(evictionsData)
+    })
+} else {
+  // if we don't have the cached file, let's go ahead
+  // and download the bigger one, manipulate it with some
+  // function processDataAndReduceItsSize, then save it to the server
+  // so the next time we run this code chunk, we'll just load the 
+  // cached (smaller) version.
+  fetch('http://...')
+    .then((r) => r.json())
+    .then(processDataAndReduceItsSize)
+    .then((finalData) => {
+      evictionsData = finalData
+      return iodide.file.save(evictionsData, 'dataset.csv', d3.csvFormat)
```

Comment:
hmm, I need to think more about this example, because if I don't own this notebook, I'm not going to be saving the data.